cnxdb/triggers.py

```diff
@@ -0,0 +1,666 @@
+# -*- coding: utf-8 -*-
+# ###
+# Copyright (c) 2013, Rice University
+# This software is subject to the provisions of the GNU Affero General
+# Public License version 3 (AGPLv3).
+# See LICENCE.txt for details.
+# ###
+"""Database models and utilities."""
+import os
+import logging
+
+import cnxdb
+
+from cnxarchive.transforms import (
+    produce_cnxml_for_module, produce_html_for_module,
+    transform_abstract_to_cnxml, transform_abstract_to_html,
+    )
+# from .utils import split_ident_hash, IdentHashMissingVersion
+
+here = os.path.abspath(os.path.dirname(__file__))
+CNXDB_DIRECTORY = os.path.abspath(os.path.dirname(cnxdb.__file__))
+SQL_DIRECTORY = os.path.join(CNXDB_DIRECTORY, 'archive-sql')
+
+logger = logging.getLogger('cnxarchive')
+
+
+def set_version(portal_type, legacy_version, td):
```

Comment:
We'll pull them off in the next batch once we test these within this package rather than cnx-archive.